src/views/RequestQueueView.js

```diff
@@ -254,25 +208,49 @@ class RequestQueueView extends React.Component {
       (isNotYetFilled(request) && !isPageRequest(request));
   }
 
+  handleToggleSection = ({ id }) => {
+    const { accordions } = this.state;
+
+    this.setState({
+      accordions: {
+        ...accordions,
+        [id]: !accordions[id],
+      },
+    });
+  }
+
   render() {
     const {
       isLoading,
       onClose,
       data: {
-        item,
         request,
       },
     } = this.props;
     const {
       requests,
       confirmMessage,
       error,
+      accordions,
     } = this.state;
     const count = requests.length;
-    const { title } = request?.instance || {};
-    const itemLink = (item.barcode)
-      ? (<Link to={`/inventory/view/${get(request, 'instanceId')}/${get(request, 'holdingsRecordId')}/${item.id}`}>{item.barcode}</Link>)
-      : '-';
+    const {
+      title,
+      publication,
+      contributors,
+    } = request?.instance || {};
+    const notYetFilledRequests = [];
+    const inProgressRequests = [];
+
+    requests.forEach(r => {
```

Comment:
I believe it should be done once and should be stored in the state

Reply:
Yeah. I believe we should extract this logic from component. It will be done in scope of migration to new queue end-points. I am working on it right now.